### PR TITLE
Features/build system

### PR DESCRIPTION
**Pull request by abdheshnayak**
Added feature BuildJobs with git registries.

### CODE REVIEW

File: apps/container-registry/internal/app/git-webhook-consumer.go [hunk 1]
```diff
@@ -121,73 +125,105 @@ func invokeProcessGitWebhooks(d domain.Domain, consumer kafka.Consumer, producer
...
 			if err := yaml.Unmarshal(b, &m); err != nil {
 				return err
 			}
 
 			b1, err := json.Marshal(t.AgentMessage{
-				AccountName: build.AccountName,
+				AccountName: build.Spec.AccountName,
 				// AccountName: "kl-core-registry",
-				ClusterName: "kl-registry-859874",
+				ClusterName: "sample-359704",
```
Comment (nxtcoder17) [inline apps/container-registry/internal/app/git-webhook-consumer.go:226]: this clusterName, should be a parameter, not a constant value

File: apps/container-registry/internal/app/git-webhook-consumer.go [hunk 2]
```diff
@@ -196,7 +232,7 @@ func invokeProcessGitWebhooks(d domain.Domain, consumer kafka.Consumer, producer
 			}
 
 			po, err := producer.Produce(ctx, "kl-send-messages-to-target-wait-queue", b1, kafka.MessageArgs{
-				Key: []byte(build.AccountName),
+				Key: []byte(build.Spec.AccountName),
 				Headers: map[string][]byte{
 					// "topic": []byte(common.GetKafkaTopicName(build.AccountName, "kl-registry-859874")),
 					"topic": []byte(common.GetKafkaTopicName("kl-core-registry", "kl-registry-859874")),
```
Comment (nxtcoder17) [inline apps/container-registry/internal/app/git-webhook-consumer.go:238]: value for "topic", should be a parameter, not a constant value

File: apps/container-registry/internal/app/main.go [hunk 3]
```diff
@@ -247,7 +254,13 @@ var Module = fx.Module("app",
 				},
 			})
 
-			return b_auth(c)
+			r := b_auth(c)
+
+			if c.Response().StatusCode() == 401 {
+				log.Println("Unauthorized", path, method, _user, _pass)
```
Comment (nxtcoder17) [inline apps/container-registry/internal/app/main.go:260]: use our logging.Logger implementation instead

File: apps/container-registry/internal/app/graph/struct-to-graphql/build.graphqls [hunk 4]
```diff
@@ -10,10 +8,9 @@ type Build @shareable {
   markedForDeletion: Boolean
   name: String!
   recordVersion: Int!
-  repository: String!
   source: Kloudlite_io__apps__container___registry__internal__domain__entities_GitSource!
+  spec: Github_com__kloudlite__operator__apis__distribution__v1_BuildRunSpec!
```
Comment (nxtcoder17) [inline apps/container-registry/internal/app/graph/struct-to-graphql/build.graphqls:12]: re run `task k8s-schema`, the naming pattern has actually been updated

now, this `Kloudlite_io__apps__container___registry__internal__domain__entities_GitSource` would be updated into this `Kloudlite__io___apps___container____registry___internal___domain___entities__GitSource`

tldr: one more underscore in everything


then, issue `task gql`, and then also fix the resolvers

File: apps/container-registry/internal/app/main.go [hunk 5]
```diff
@@ -220,9 +219,17 @@ var Module = fx.Module("app",
...
+			if method == "HEAD" {
+				return c.Next()
+			}
+
+			var _user, _pass string
+
 			b_auth := basicauth.New(basicauth.Config{
 				Realm: "Forbidden",
 				Authorizer: func(u string, p string) bool {
+					_user = u
```
Comment (nxtcoder17) [inline apps/container-registry/internal/app/main.go:231]: should remove the _ prefix from _user, and _pass

File: apps/container-registry/internal/domain/main.go [hunk 6]
```diff
@@ -152,18 +156,25 @@ func (d *Impl) ProcessRegistryEvents(ctx context.Context, events []entities.Even
...
+				_, err := d.repositoryRepo.Create(ctx, &entities.Repository{})
 				if err != nil {
-					log.Println(err)
+					return err
 				}
+				return nil
+			}
+
+			ee.LastUpdatedBy = common.CreatedOrUpdatedBy{
+				UserName: e.Actor.Name,
```
Comment (nxtcoder17) [inline apps/container-registry/internal/domain/main.go:172]: actually, fill the other remaining fields in common.CreatedOrUpdatedBy type

File: apps/wg-proxy/.gitignore [hunk 7]
```diff
@@ -1 +1 @@
-main
+conf.json
```
Comment (nxtcoder17) [inline apps/wg-proxy/.gitignore:1]: but, conf.json is already listed in git

File: go.mod [hunk 8]
```diff
@@ -52,6 +52,10 @@ require (
 )
 
 require (
+<<<<<<< HEAD
```
Comment (nxtcoder17) [inline go.mod:55]: resolve these conflicts, these are causing CodeQL failure

**Review by nxtcoder17 — CHANGES_REQUESTED** [review]

File: apps/container-registry/internal/app/graph/struct-to-graphql/build.graphqls (hunk 4 above)
Comment (abdheshnayak) [reply apps/container-registry/internal/app/graph/struct-to-graphql/build.graphqls]: Fixed

File: apps/container-registry/internal/app/main.go (hunk 5 above)
Comment (abdheshnayak) [reply apps/container-registry/internal/app/main.go]: Fixed

File: apps/wg-proxy/.gitignore (hunk 7 above)
Comment (abdheshnayak) [reply apps/wg-proxy/.gitignore]: Fixed

File: go.mod (hunk 8 above)
Comment (abdheshnayak) [reply go.mod]: Fixed

**Review by nxtcoder17 — APPROVED** [review]
